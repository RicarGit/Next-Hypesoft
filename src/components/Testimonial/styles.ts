import styled from "styled-components"

export const TestimonialSection = styled.section`
  width: 100%;
  position: absolute;
  top: 4661px;
  gap: 50px;
`

export const Container = styled.div`
  width: 1320px;
  height: 528px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
`

export const TestimonialContainer = styled.div`
  width: inherit;
  height: 411px;
  display: flex;
  gap: 44px;
`

export const QuoteContainer = styled.div`
  position: relative;
  width: 964px;
  height: inherit;
  padding: 50px;
  border-radius: 3px;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3),
    rgba(223, 227, 235, 0.3)),
    #FFFFFF;

  &::before {
    content: '';
    width: 32px;
    height: 32px;
    display: block;
    position: absolute;
    border-radius: 4px;
    transform: rotate(45deg);
    top: 38px;
    left: -15px;
    z-index: -1;
    background: linear-gradient(0deg, rgba(223, 227, 235, 0.3),
      rgba(223, 227, 235, 0.3)),
      #FFFFFF;
  }
`

export const Rating = styled.div`
  display: flex;
  align-items: center;
  font-size: 18px;
  font-weight: 400;
  line-height: 28px;
  gap: 8px;
  color: ${({ theme }) => theme.colors.primary};
`

export const Quote = styled.q`
  width: 864px;
  height: 255px;
  display: block;
  padding-top: 26px;
  font-size: 32px;
  font-weight: 400;
  line-height: 51.2px;
  color: ${({ theme }) => theme.colors.primary};
`

export const QuoteImageWrapper = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  padding: 24px;
`

export const ClientContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 4px;
`