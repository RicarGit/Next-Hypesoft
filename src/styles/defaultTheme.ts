import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme