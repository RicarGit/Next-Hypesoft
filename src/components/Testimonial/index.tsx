import * as S from './styles'

import SectionHeader from 'components/shared/SectionHeader'
import Image from 'next/image'
import fullStarRating from './fullStarRating.svg'
import quote from './quote.svg'

const Testimonial = () => {
  return (
    <S.TestimonialSection>
      <S.Container>

        <SectionHeader>Testimonial</SectionHeader>
        <S.TestimonialContainer>
          <S.ClientContainer>

            <S.Client className='active'>
              <S.ClientAvatar />
              <S.ClientInfoContainer>

                <S.ClientName>
                  Sundar Pichai
                </S.ClientName>

                <S.ClientOccupation>
                  CEO & Founder of <span>Google</span>
                </S.ClientOccupation>

              </S.ClientInfoContainer>
            </S.Client>

            <S.Client>
              <S.ClientAvatar />
              <S.ClientInfoContainer>

                <S.ClientName>
                  Shantanu Narayen
                </S.ClientName>

                <S.ClientOccupation>
                  CEO of <span>Adobe</span>
                </S.ClientOccupation>

              </S.ClientInfoContainer>
            </S.Client>

            <S.Client>
              <S.ClientAvatar />
              <S.ClientInfoContainer>

                <S.ClientName>
                  Noah Levin
                </S.ClientName>

                <S.ClientOccupation>
                  Director of Design, <span>Figma</span>
                </S.ClientOccupation>

              </S.ClientInfoContainer>
            </S.Client>

            <S.Client>
              <S.ClientAvatar />
              <S.ClientInfoContainer>

                <S.ClientName>
                  Stewart Butterfield
                </S.ClientName>

                <S.ClientOccupation>
                  CEO & Founder of <span>Slack</span>
                </S.ClientOccupation>

              </S.ClientInfoContainer>
            </S.Client>

          </S.ClientContainer>
          <S.QuoteContainer>

            <S.Rating>
              <Image src={fullStarRating} />
              5.0 Star Rating
            </S.Rating>

            <S.Quote>
              If you're looking for someone who will challenge your UX/UI
              thinking and really cut to the core of what's important for users,
              then Jesse is your man. On top of that, he brings a level of
              enthusiasm to the craft that's energizing for everyone who works with him.
            </S.Quote>

            <S.QuoteImageWrapper>
              <Image src={quote} />
            </S.QuoteImageWrapper>

          </S.QuoteContainer>
        </S.TestimonialContainer>

      </S.Container>
    </S.TestimonialSection>
  )
}

export default Testimonial