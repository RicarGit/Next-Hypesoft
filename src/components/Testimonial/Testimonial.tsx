import * as S from './Testimonial.styles'

import Image from 'next/image'
import { Assets } from '~/assets'

import { SectionHeader } from '~/shared/SectionHeader'
import { TestimonialClient } from '~/components/TestimonialClient'

export const Testimonial = () => {
  return (
    <S.TestimonialSection>
      <S.Container>
        <SectionHeader>Testimonial</SectionHeader>

        <S.TestimonialContainer>
          <S.ClientContainer>
            <TestimonialClient
              name='Sundar Pichai'
              occupation='CEO & Founder of'
              company='Google'
              className='active'
            />
            <TestimonialClient
              name='Shantanu Narayen'
              occupation='CEO of'
              company='Adobe'
            />
            <TestimonialClient
              name='Noah Levin'
              occupation='Director of Design,'
              company='Figma'
            />
            <TestimonialClient
              name='Stewart Butterfield'
              occupation='CEO & Founder of'
              company='Slack'
            />
          </S.ClientContainer>

          <S.QuoteContainer>
            <S.Rating>
              <Image src={Assets.FullStarRating} width={32} height={32} alt='full star' />
              5.0 Star Rating
            </S.Rating>

            <S.Quote>
              If you're looking for someone who will challenge your UX/UI
              thinking and really cut to the core of what's important for users,
              then Jesse is your man. On top of that, he brings a level of
              enthusiasm to the craft that's energizing for everyone who works with him.
            </S.Quote>

            <S.QuoteImageWrapper>
              <Image src={Assets.Quote} width={96} height={76} alt='big quote' />
            </S.QuoteImageWrapper>
          </S.QuoteContainer>
        </S.TestimonialContainer>
      </S.Container>
    </S.TestimonialSection>
  )
}