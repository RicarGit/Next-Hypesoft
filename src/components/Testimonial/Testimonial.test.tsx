import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import Testimonial from '.'

describe('Testimonial', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Testimonial />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})