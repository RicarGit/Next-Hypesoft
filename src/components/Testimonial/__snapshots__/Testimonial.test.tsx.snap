// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonial should match snapshot 1`] = `
<section
  class="sc-bczRLJ igNEoN"
>
  <div
    class="sc-gsnTZi Ixjja"
  >
    <div
      class="sc-ftvSup dtPkWK"
    >
      <h2
        class="sc-papXJ ktRCrt"
      >
        Testimonial
      </h2>
    </div>
    <div
      class="sc-dkzDqf kyRXty"
    >
      <div
        class="sc-iBkjds fTArYX"
      >
        <div
          class="sc-jqUVSM cWDEAv active"
        >
          <div
            class="sc-kDDrLX ehnXY"
          />
          <div
            class="sc-iqcoie tWTiC"
          >
            <p
              class="sc-crXcEl jcqmjt"
            >
              Sundar Pichai
            </p>
            <p
              class="sc-evZas kjUWrR"
            >
              CEO & Founder of
               
              <span>
                Google
              </span>
            </p>
          </div>
        </div>
        <div
          class="sc-jqUVSM cWDEAv"
        >
          <div
            class="sc-kDDrLX ehnXY"
          />
          <div
            class="sc-iqcoie tWTiC"
          >
            <p
              class="sc-crXcEl jcqmjt"
            >
              Shantanu Narayen
            </p>
            <p
              class="sc-evZas kjUWrR"
            >
              CEO of
               
              <span>
                Adobe
              </span>
            </p>
          </div>
        </div>
        <div
          class="sc-jqUVSM cWDEAv"
        >
          <div
            class="sc-kDDrLX ehnXY"
          />
          <div
            class="sc-iqcoie tWTiC"
          >
            <p
              class="sc-crXcEl jcqmjt"
            >
              Noah Levin
            </p>
            <p
              class="sc-evZas kjUWrR"
            >
              Director of Design,
               
              <span>
                Figma
              </span>
            </p>
          </div>
        </div>
        <div
          class="sc-jqUVSM cWDEAv"
        >
          <div
            class="sc-kDDrLX ehnXY"
          />
          <div
            class="sc-iqcoie tWTiC"
          >
            <p
              class="sc-crXcEl jcqmjt"
            >
              Stewart Butterfield
            </p>
            <p
              class="sc-evZas kjUWrR"
            >
              CEO & Founder of
               
              <span>
                Slack
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="sc-hKMtZM dHdjG"
      >
        <div
          class="sc-eCYdqJ gxuMRn"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="full star rating"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          5.0 Star Rating
        </div>
        <q
          class="sc-jSMfEi ca-Dggb"
        >
          If you're looking for someone who will challenge your UX/UI thinking and really cut to the core of what's important for users, then Jesse is your man. On top of that, he brings a level of enthusiasm to the craft that's energizing for everyone who works with him.
        </q>
        <div
          class="sc-gKXOVf jaKzGq"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="big quotes"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
